collated/main/tby1994.md

```diff
@@ -0,0 +1,884 @@
+    }
+}
+```
+###### /java/seedu/address/logic/LogicManager.java
+``` java
+    @Override
+    public CommandResult execute(String commandText) throws CommandException, ParseException {
+        logger.info("----------------[USER COMMAND][" + commandText + "]");
+        try {
+            Command command = addressBookParser.parseCommand(commandText, model.getCommandMode());
+            command.setData(model, history, undoRedoStack);
+            CommandResult result = command.execute();
+            undoRedoStack.push(command);
+            return result;
+        } finally {
+            history.add(commandText);
+        }
+    }
+```
+###### /java/seedu/address/logic/parser/AddressBookParser.java
+``` java
+/**
+ * Parses user input.
+ */
+public class AddressBookParser {
+
+
+    private static final String MODE_ADDRESS_BOOK = "addressbook ab";
+    private static final String MODE_TASK_MANAGER = "taskmanager tm";
+
+    /**
+     * Used for initial separation of command word and args.
+     */
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Command parseCommand(String userInput, String commandMode) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord").toLowerCase();
+        final String arguments = matcher.group("arguments");
+
+        if (StringUtil.containsWordIgnoreCase(MODE_ADDRESS_BOOK, commandMode)) {
+            switch (commandWord) {
+
+            case AddCommand.COMMAND_WORD:
+                return new AddCommandParser().parse(arguments);
+
+            case ChangeModeCommand.COMMAND_WORD:
+                return new ChangeModeCommandParser().parse(arguments);
+
+            case EditCommand.COMMAND_WORD:
+                return new EditCommandParser().parse(arguments);
+
+            case DeleteCommand.COMMAND_WORD:
+                return new DeleteCommandParser().parse(arguments);
+
+            case FindCommand.COMMAND_WORD:
+                return new FindCommandParser().parse(arguments);
+
+            case SelectCommand.COMMAND_WORD:
+                return new SelectCommandParser().parse(arguments);
+
+            case ListCommand.COMMAND_WORD:
+                return new ListCommand();
+
+            case DetagCommand.COMMAND_WORD:
+                return new DetagCommandParser().parse(arguments);
+
+            default:
```

Comment:
fall through are generally not recommended too. leave a comment there if its intended.